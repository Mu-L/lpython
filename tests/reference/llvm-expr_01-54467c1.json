{
    "basename": "llvm-expr_01-54467c1",
    "cmd": "lpython --no-color --show-llvm {infile} -o {outfile}",
    "infile": "tests/expr_01.py",
    "infile_hash": "4284fe3a1b4dd3e5d1de1357a79e9a25b426ca245b4cc91cf99e8547",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm-expr_01-54467c1.stdout",
    "stdout_hash": "62370056084f1f3bef4e9ba8815234382ab27a399d0873136f1be02d",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}