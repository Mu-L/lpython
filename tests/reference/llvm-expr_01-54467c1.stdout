; ModuleID = 'LFortran'
source_filename = "LFortran"

@0 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@1 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1

define void @__module___main_____main__global_stmts() {
.entry:
  call void @__module___main___main0()
  br label %return

return:                                           ; preds = %.entry
  ret void
}

define void @__module___main___main0() {
.entry:
  %x = alloca i32, align 4
  %x2 = alloca i64, align 8
  %y = alloca float, align 4
  %y2 = alloca double, align 8
  store i32 25, i32* %x, align 4
  %0 = load i32, i32* %x, align 4
  %1 = sext i32 %0 to i64
  %2 = call i8* (i32, i8*, ...) @_lcompilers_string_format_fortran(i32 2, i8* null, i32 2, i64 %1)
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @1, i32 0, i32 0), i8* %2, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @0, i32 0, i32 0))
  br label %return

return:                                           ; preds = %.entry
  ret void
}

declare i8* @_lcompilers_string_format_fortran(i32, i8*, ...)

declare void @_lfortran_printf(i8*, ...)

define i32 @main(i32 %0, i8** %1) {
.entry:
  call void @_lpython_call_initial_functions(i32 %0, i8** %1)
  call void @__module___main_____main__global_stmts()
  call void @_lpython_free_argv()
  br label %return

return:                                           ; preds = %.entry
  ret i32 0
}

declare void @_lpython_call_initial_functions(i32, i8**)

declare void @_lpython_free_argv()
